server/src/main/java/pl/kejbi/tin/socket/Server.java

```diff
@@ -0,0 +1,43 @@
+package pl.kejbi.tin.socket;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+
+@Getter
+@Setter
+public class Server extends Thread {
+    private ServerSocket serverSocket;
+    private volatile boolean stop;
+    private Logger logger = LoggerFactory.getLogger(Server.class);
+
+    public Server(ServerSocket socket) throws SocketException {
+        serverSocket = socket;
+    }
+
+    @Override
+    public void run() {
+        while(!stop) {
+            try {
+                Socket clientSocket = serverSocket.accept();
```

Comment:
Nie zauważyłem, abyś ustawił timeout na socket nasłuchujący. Wydaje mi się, że serwer w tym wypadku w ogóle nie odpowiada na sygnały zewnętrzne (oprócz SIGTERM oczywiście), bo się całkowicie blokuje.

Reply:
moim zdaniem powinien się blokować. U Kacpra serwer zapewne będzie puszczany na swoim watku i będzie musiał czekać na przychodzace wiadomości. Gdy ta wiadomość przyjdzie właczy on nowego threada z handlerem `new Thread(clientHandler).start();` i znowu będzie czekał na nowa wiadomość. Nie jestem pewien czy w ogóle zasadne jest nawet kiedykolwiek zatrzymywanie serwera bo on i tak zawsze musi pracować przy działaniu aplikacji

Reply:
teraz myślę sobie, że chyba ten stop nie ma sensu bo i tak server zablokuje się na accept i wyłączy się dopiero jak obsłuży kolejnego klienta, zrobię chyba while(true)

Reply:
Zapytam Grochowa co do tych sygnałów, bo mnie ciekawi czy można zrobić tak, że serwer nie reaguje na sygnały zewnętrzne przez występowanie blokady dopóki nie przyjdzie komunikat. Ja zrobiłem tak, że mój serwer co sekundę przestaje się blokować na chwilę jeżeli przez sekundę nie dostanie wiadomości. Plus takiego rozwiązania jest taki, że można w ładny sposób zamknąć deskryptor socketa nasłuchującego. U prowadzącego jest takie zdanie:
> Niedopuszczalne jest, aby jakikolwiek wątek "zawieszał" się w oczekiwaniu na cokolwiek bez możliwości przerwania tego oczekiwania na żądanie.

Myślę, że to dotyczy również i socketa nasłuchującego oprócz innych, efemerycznych.

Reply:
Znalazłem też takie zdanie:
> Wyłączanie systemu rozproszonego powinno moc przebiec w elegancki i kontrolowany sposób.

Reply:
czyli po prostu dodać timeouta do serwera